src/jws/sign.rs

```diff
@@ -241,7 +280,197 @@ mod tests {
 
             assert_eq!(
                 format!("{}", err),
-                "Unsupported crypto or method: Unsuported signature type"
+                "Unsupported crypto or method: Unsupported signature type"
+            );
+        }
+    }
+
+    #[test]
+    fn sign_compact_works() {
+        _sign_compact_works::<Ed25519KeyPair>(
+            ALICE_KID_ED25519,
+            ALICE_KEY_ED25519,
+            ALICE_PKEY_ED25519,
+            "example-typ-1",
+            Algorithm::EdDSA,
+            PAYLOAD,
+        );
+
+        _sign_compact_works::<P256KeyPair>(
+            ALICE_KID_P256,
+            ALICE_KEY_P256,
+            ALICE_PKEY_P256,
+            "example-typ-2",
+            Algorithm::Es256,
+            PAYLOAD,
+        );
+
+        _sign_compact_works::<K256KeyPair>(
+            ALICE_KID_K256,
+            ALICE_KEY_K256,
+            ALICE_PKEY_K256,
+            "example-typ-3",
+            Algorithm::Es256K,
+            PAYLOAD,
+        );
+
+        fn _sign_compact_works<K: FromJwk + KeySign + KeySigVerify>(
+            kid: &str,
+            key: &str,
+            pkey: &str,
+            typ: &str,
+            alg: Algorithm,
+            payload: &str,
+        ) {
+            let res = _sign_compact::<K>(kid, key, typ, alg.clone(), payload);
+
+            let msg = res.expect("Unable _sign_compact");
+
+            let mut buf = vec![];
+            let msg = jws::parse_compact(&msg, &mut buf).expect("Unable parse_compact.");
```

Comment:
Let's look to messages examples

- "Unable parse_compact."
- "Unable _sign_compact"
- "unable from_jwk"

Could you make them consistent? I suggest "Unable _sign_compact" form.

Reply:
Done.